.github/workflows/pr.yml

```diff
@@ -5,8 +5,57 @@ on:
 
 permissions: read-all
 
+env:
+  MIN_PYTHON_VERSION: "3.10"
+
 jobs:
-  tests:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@755da8c3cf115ac066823e79a1e1788f8940201b  # v3
+      - uses: actions/setup-python@5ccb29d8773c3f3f653e1705f474dfaa8a06a912  # v4
+        with:
+          python-version: ${{ env.MIN_PYTHON_VERSION }}
+      - name: pre-commit
+        uses: pre-commit/action@646c83fcd040023954eafda54b4db0192ce70507  # v3
+  mypy:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@755da8c3cf115ac066823e79a1e1788f8940201b  # v3
+    - uses: actions/setup-python@5ccb29d8773c3f3f653e1705f474dfaa8a06a912  # v4
+      with:
+        python-version: ${{ env.MIN_PYTHON_VERSION }}
+    - name: Install pipenv
+      run: |
+        python -m pip install --no-cache-dir --upgrade pipenv
+    - name: Install dependencies
+      run: |
+        pipenv --python ${{ env.MIN_PYTHON_VERSION }}
+        pipenv install --dev
```

Comment:
@gruebel Wouldn't it be nicer to just have mypy run after the pre-commit? This whole setup is sooooo long

Reply:
I actually did at the beginning, but I had to re-add the dependencies from the Pipfile to the pre-commit config. pre-commit doesn't use the  local dependencies it creates its own venv for each hook. but let me try something out, before I say it doesn't work at all 🧐

Reply:
it is possible, but I still need to do the whole setup with installing pipenv and the projects dependencies, because pre-commit doesn't rely on the projects dependencies.